server/routes/index.js

```diff
@@ -105,70 +109,92 @@ router.get("/callback", async function (req, res, next) {
         let topArtists;
 
         try {
-          topArtists = await getSpotifyTopArtists(access_token);
-          const { artistEventRequestArray, listOfArtists } = topArtists;
-
-          try {
-            const festivalResults = await getBandsInTownEvents(
-              artistEventRequestArray,
-              listOfArtists
-            );
+          const userDetailsResponse = await axios.get(
+            "https://api.spotify.com/v1/me",
+            {
+              headers: {
+                Authorization: "Bearer " + access_token,
+              },
+            }
+          );
 
-            try {
-              const userDetailsResponse = await axios.get(
-                "https://api.spotify.com/v1/me",
-                {
-                  headers: {
-                    Authorization: "Bearer " + access_token,
-                  },
-                }
-              );
+          if (userDetailsResponse.status === 200) {
 
-              if (userDetailsResponse.status === 200) {
+            // Successfully logged in
+            hashedId = md5(userDetailsResponse.data.id);
+            
 
-                // Successfully logged in
-                hashedId = md5(userDetailsResponse.data.id);
-                const user = {
-                  userId: hashedId,
-                };
+            // Set user's session
+            req.session.user = {
+              access_token: access_token,
+              refresh_token: refresh_token,
+              userId: hashedId,
+            }
+            console.log("successfully logged in")
+
+            const user = await User.findOneAndUpdate({ userId: hashedId }, 
+              {
+                userId: hashedId
+              }, {
+              new: true,
+              upsert: true,
+            });
+
+            const currentSpotifyImports = await SpotifyImports.find({ userId: user.userId }).sort({ timeOfImport: -1 }).limit(1);
+            if (currentSpotifyImports.length > 0 && (new Date()).getTime() - currentSpotifyImports[0].timeOfImport.getTime() < 24 * 60 * 60 * 1000) {
+              throw new Error('no need to update imports, last import was done less than a day ago');
```

Comment:
Had to remove the throw otherwise the rest of the stuff will not run, I added to Azra's FE branch

Reply:
LGTM otherwise

Reply:
Were you connecting to atlas when running this? Or your local db?

If we remove the throw, then we would be importing from spotify and caching the spotify data on every login. We want to only story it once every 24 hours though right?